.clang-format

```diff
@@ -0,0 +1,245 @@
+---
```

Comment:
Is this just a dump of the format that was previously hardcoded in the helper script?

Reply:
Yes, generated by `clang-format --style="{BasedOnStyle: Mozilla, IndentWidth: 4, ColumnLimit: 120}" --dump-config > .clang-format` using the version 18.1.5 as specified in [hatch.toml](https://github.com/DataDog/dd-trace-py/blob/2f7063bc3cb3ddacbc97c0e53627944d63bcf24b/hatch.toml#L27)